### DISCUSSION TITLE
Episode 142: 🇸🇮 SRT23 - Lake Bled & Bled Cake + Haskell, Rust & C++

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#33] [Episode]
Episode Link: https://adspthepodcast.com/2023/08/11/Episode-142.html

Feel free to write comments or ask questions 🙂

**Comment by John-LittleBearLabs** [comment]
As to why it doesn't call `chars()` _for_ you, it doesn't want to make assumptions about what you mean.

https://doc.rust-lang.org/std/primitive.str.html
> String slices are always valid UTF-8.

What exactly does it mean to iterate over UTF-8 ? You can specify what you mean with a method.

https://doc.rust-lang.org/std/primitive.str.html#method.chars
> [char](https://doc.rust-lang.org/std/primitive.char.html) represents a Unicode Scalar Value

https://doc.rust-lang.org/std/primitive.str.html#method.bytes
> iterate through a string slice by byte

And by 'byte' they mean the code units, aka 8-bit octets. Rust terminology often makes a reasonable CHAR_BIT assumption.

There could be another approach.
> Iteration over grapheme clusters... is not provided by Rust’s standard library, check crates.io

**Reply by codereport** [reply]
Thanks for the info! I will try to remember to mention this the next time we record an episode : )
